### DISCUSSION TITLE
PR review comments not showing up in "Files changed" (and are not "Outdated")

### DISCUSSION THREAD

**Discussion by alexjpwalker** [community/community#30638] [Pull Requests]
Generally, when writing a PR review comment on a specific line of code, it becomes visible to anyone viewing the "Conversation" or "Files changed" tabs of the PR. In "Files changed", the comment shows up immediately below the line(s) of code it was written against.

However, posted comments intermittently do not show up at all. We know this occurs if the comment goes "Outdated" due to code changes, but this is not happening in this case - the comment simply disappears without trace, with no further code changes.

One common scenario I've experienced (for example) is, when adding a review comprising multiple comments, the first one I write sometimes fails to show up in "Files changed". It exists, and can be accessed from the small "Conversations" button on the "Files changed" tab, but that is a significantly worse UX than seeing the thread at the relevant code line - highly frustrating during the review process.

**Comment by AWqxKAWERbXo** [comment]
It also looks like comments are only visible to me as a reviewer, as long as i don't finish my review. 

This results in bad UX because while I thought, my comments and code change suggestions are visible to the PR creator, in fact, they were only visible to me, as long as I don't finish my review. 

The drafted comments should be somehow be different visually (grayed out, not 100% opaque, or a flag / badge) as long as the review was not submitted.

**Reply by Rimadaqch** [reply]
comment

**Reply by kroggen** [reply]
That is a good idea, to show the comments with a different color somewhere while the review was not submitted. Or display that they are not yet visible to other users.

In our company we had problems related to this twice, because people do not know that single comments are not appearing to others while they don't click on finish review, but their thought is "I will finish the review once the PR creator answers this question" but the question is not visible to others!!!

**Reply by kroggen** [reply]
The current UX is not intuitive! Just on our company 2 different developers already did the exact same thing described above.

They write a question linked to code and instead of clicking on "Add a single comment" they click on "Start a review", but it is not directly clear that the written messages are only visible to themselves!

This causes misunderstandings and make us lose a lot of time

**Reply by Lasse-BeCause** [reply]
Agreed, very confusing, as there is nothing that indicates that the comment is only visible to the reviewer.

**Reply by PureGero** [reply]
Agreed. I just did the same thing and spent ages trying to debug why my coworker couldn't see my comment.

**Reply by guythiebautUL** [reply]
Just happened to me and I had to have a screen share with the reviewee to confirm that they could not see the comments - then I saw and clicked on the submit review button - I agree the two different ways of adding comments/reviews is not at all obvious.

**Reply by mihawk90** [reply]
wow... this comment probably just saved years of my sanity.

I was wondering why noone was responding to a review comment I posted months ago, and it turns out I had no idea the review had to be "finished" ... I thought I was just writing a comment, WTH.

**Reply by Fredolx** [reply]
Thanks, I tortured my poor contributors twice now thinking it was user error, well the user error was on me! I agree it's poor UX and the comments should be dimmed (less opaque) until the review is submitted to make it clear the comments are not visible.

**Comment by michaelloo** [comment]
I experienced exactly the same problem as the original post:

> One common scenario I've experienced (for example) is, when adding a review comprising multiple comments, the first one I write sometimes fails to show up in "Files changed". It exists, and can be accessed from the small "Conversations" button on the "Files changed" tab, but that is a significantly worse UX than seeing the thread at the relevant code line - highly frustrating during the review process.

And I had submitted my review, so other comments from the same review would be shown in "Files changed" but not the first one and it was not marked as "Outdated" and no code was updated.

I'm not sure if it is related but the Sys Engineers in my company had updated GHE from 3.8.8 to 3.8.10 a week or two before this started happening.

**Comment by Zebiano** [comment]
I'm also experiencing the same issue: comments appearing in the "Conversation" tab, but not all of them appearing on the "Files changed" tab.

Highly frustrating, as I'd go through all the comments on the "Files changed" tab (since they're easier to grasp there because of having the rest of the code as context), only to go back to the "Conversation" tab when I'm done and realise there are still comments left unresolved.

The weird part is when in the "Files changed" tab, there's a small "Conversations" button at the top that lists all the comments left unresolved. If you click on one, you'll automatically get scrolled to where the comment is. But when you try to click on a comment that is not showing up, clicking it does absolutely nothing. But it does exist, as it's part of the list of comments left. So there's definitely something wrong here.

**Comment by NoureddinBakir** [comment]
Issue is persistent with one of my PR's as well. I saw my coworker's screen as having placed the comments but even after clearing cookies, cache, logging out and in numerous times the comments are no where to be found.

**Comment by JetFault** [comment]
Same is happening for us on GHE Server 3.8

**Comment by xepelinalf** [comment]
I have the exact same problem; I cannot see the responses to my own comments. How is this still happening?!

**Reply by MatheusIam** [reply]
As far as I can tell, it seems like you need to finalize the review based on the information provided above

> It also looks like comments are only visible to me as a reviewer, as long as i don't finish my review.

**Reply by KingDuckZ** [reply]
Everyone is posting this "finish my review" thing, what does it mean in practice? How does one "finish their review"??

**Comment by nazreen** [comment]
Just realised the past few times when my dev said " I can't see the comments" it was due to this too. 

please fix.

**Comment by Daerst** [comment]
I kinda see that for larger PRs it might be worth adding _all_ the comments before showing them to the developer in question, so that they don't start reworking code based on incomplete information. Still, I think this needs to be made clearer to the commenter.

**Comment by filipbatur** [comment]
Had this happen to us on several occasions, the UX is completely misleading.

**Comment by samakshkhatri-crest** [comment]
This process of "starting" and "finishing" a review seems does not intuitive at all. As a reviewer, my review should be considered "finished" when I provide an approval. 

Also it's unusual that PR authors cannot approve their own authored PR even if they are added as a reviewer and the assignee is someone else. This should take precedence over author restrictions. Neither Gitlab or Bitbucket restrict this inherently, of course approval rules can be configured separately but those are optional.

**Reply by RazorBach** [reply]
> This process of "starting" and "finishing" a review seems does not intuitive at all. As a reviewer, my review should be considered "finished" when I provide an approval.
> 
> Also it's unusual that PR authors cannot approve their own authored PR even if they are added as a reviewer and the assignee is someone else. This should take precedence over author restrictions. Neither Gitlab or Bitbucket restrict this inherently, of course approval rules can be configured separately but those are optional.

+1. github please consider improving this.

**Comment by RazorBach** [comment]
For anyone expericing similar issue. Please follow the steps in [the offical guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request) to send out review comments.
especially [step 10](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request#:~:text=When%20you%27re%20done%2C%20click%20Start%20a%20review)

> When you're done, click Start a review. If you have already started a review, you can click Add review comment.
Before you submit your review, your line comments are pending and only visible to you. You can edit pending comments anytime before you submit your review. To cancel a pending review, including all of its pending comments, click Review changes above the changed code, then click Abandon review.

**Reply by KingDuckZ** [reply]
I can't believe posting a comment needs a tutorial in 2025. This website's got "Microsoft quality" stamped all over. I only found out because my colleague addressed comments from other people but not mine and I finally decided to share a screenshot with him pressing for an answer. OP reported this in 2022, it's unbelievable such a prominent bug is still there 3 years later!

**Comment by jet4419** [comment]
You must click the "Review changes" then click the "Submit review" to see your comments by other members of the team

**Reply by KingDuckZ** [reply]
Could you be more specific? I have an "Add your review" green button at the top of the page which I click and seems to do nothing - it just loads a new page showing the combined diff, then at the bottom I have "Merge pull request" and "Close pull request". Ctrl-F doesn't find any "submit" string at all

**Reply by jet4419** [reply]
Here's the screenshot after clicking the "Review changes" button

![Screenshot_20250602_185412](https://github.com/user-attachments/assets/ee31cfae-f5e7-4eaa-bcb3-a91bb3024d61)

**Comment by basantkr762** [comment]
To fix PR review comments not showing in "Files changed" (#30638):

Refresh Page: Reload the PR page in your browser.
Check Filters: Ensure "Show outdated" or other filters aren’t hiding comments.
Verify Comment Status: Confirm comments are submitted, not pending.
Clear Cache: Clear browser cache and reload.
Check GitHub Status: Visit [www.githubstatus.com](http://www.githubstatus.com/) for platform issues.
Report to Support: If unresolved, contact GitHub Support at https://support.github.com/ with PR URL and details.

Try these steps; if comments still don’t appear, it’s likely a GitHub bug.

**Comment by sambhusankar** [comment]
Hey, i am also facing the same issue, not seeing the PR review comments made by a collaborator.

**Comment by DanielGahler** [comment]
Ran into a similar issue today, it seems that GitHub starts "collecting" the responses from the collaborator (will have a *pending* note above), at which point they are NOT sent. They still need to click "Submit Review" before everyone else can see them.

**Comment by Petercopter** [comment]
I'm the PR owner, and I'm not seeing comments made by Github Copilot in Files Changed, only in the Conversation tab. Really frustrating.

**Comment by ardrabczyk** [comment]
The issues still exists in the new GH experience and is intermittent as said in the first comment, some comments are shown in the comment panel and some aren't regardless of whether they were posted with or without going through the `Start a review` process. In the old view all comments are always shown under `Conversations` tab.

**Comment by simonpolymarket** [comment]
This is inexcusable. A code sharing and reviewing platform does not show all comments. So laughable.

**Reply by Tantael** [reply]
Problem still persists, Github review feature is completly useless.

**Reply by gold2718** [reply]
I'm seeing this (or not seeing it) today on a PR I reviewed yesterday. Where the PR author changed lines of code, there is no sign of my comment (which I would normally 'Resolve' after verifying the fix). @GitHub, what is going on?
